### PR TITLE
Update markdownlint Configuration

### PR DESCRIPTION
**Pull request by mcdonnnj**
#  #

## 🗣 Description ##

This PR updates the configuration for the `markdownlint` [pre-commit](https://pre-commit.com) hook. It adds rules to do the following:

- Enforce the ATX-closed header style
- Enforce using dashes for unordered lists
- Enforce starting ordered list elements with `1.`
- Allow the `<h1>` and `` HTML tags to permit header images

Additionally the entire configuration is migrated from JSON to YAML.

## 💭 Motivation and context ##

These rule changes help enforce some of our spoken and unspoken style preferences. The switch to using a YAML-based configuration file allows for commenting rule changes to properly document what the configuration is enforcing.

## 🧪 Testing ##

I confirmed that the rule changes were correctly enforced against a testing markdown file both before and after the YAML conversion. Automatic tests pass with no issue.

## ✅ Checklist ##

* [x] This PR has an informative and human-readable title.
* [x] Changes are limited to a single goal - _eschew scope creep!_
* [x] All relevant type-of-change labels have been added.
* [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
* [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
* [x] Tests have been added and/or modified to cover the changes in this PR.
* [x] All new and existing tests pass.

### CODE REVIEW

**Review by dav3r — APPROVED** [review]
Love it- thanks for these welcome changes! 🌈

File: .mdl_config.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+---
+
+# Default state for all rules
+default": true
```
Comment (jsf9k) [inline .mdl_config.yaml]: Is the double quote character supposed to be here?

**Review by jsf9k — APPROVED** [review]
I had one question about an extra character that may be a typo.  Otherwise this looks great!  Thanks for taking the time to make these much-needed improvements.

File: .mdl_config.yaml (hunk 1 above)
Comment (dav3r) [reply .mdl_config.yaml]: Good catch!

Comment (mcdonnnj) [reply .mdl_config.yaml]: 🦅👀 I've rebased to fix, thanks for the catch!

**Review by jsf9k — APPROVED** [review]

**Pull request merged**